editor/app/src/main/java/ca/ubc/eml/soiltopargraphy/editor/ui/main/MainFragment.kt

```diff
@@ -187,8 +228,23 @@ class MainFragment : Fragment(), OnMapReadyCallback, GoogleMap.OnMapLongClickLis
     }
 
     override fun onMapLongClick(point: LatLng) {
-        this.googleMap.addMarker(MarkerOptions()
-                .position(point)
-                .draggable(true))
+        var marker = getMarkerIconFromDrawable(resources.getDrawable(R.drawable.ic_flag))
+        if (PolyUtil.containsLocation(point, this.polygon.points, true)) {
+            this.googleMap.addMarker(MarkerOptions()
+                    .icon(marker)
+                    .snippet("Press here to remove marker")
+                    .title("MARKER")
+                    .position(point)
+                    .draggable(true))
+        }
+    }
+
+    private fun getMarkerIconFromDrawable(drawable: Drawable): BitmapDescriptor {
```

Comment:
You can replace this whole function in 1 line! Check out this link https://stackoverflow.com/questions/18053156/set-image-from-drawable-as-marker-in-google-map-version-2.